app/models/amr_count.rb

```diff
@@ -1,6 +1,32 @@
 # Model that has data fields necessary for displaying
 # anti microbial resistance (AMR) information for a
 # given pipeline run of a sample.
+#
+# Currently the following values exist for drug_family:
+#   ["AGly",
+#   "Bla",
+#   "Colistin",
+#   "Ecoli_Bla",
+#   "Fcd",
+#   "Fcyn",
+#   "Flq",
+#   "Gly",
+#   "hydrolase_Bla",
+#   "MLS",
+#   "Nitroimidazole_Gene_Ntmdz",
+#   "Oxzln",
+#   "Phe",
+#   "Rif",
+#   "Sul",
+#   "Tet",
+#   "Tmt"]
 class AmrCount < ApplicationRecord
   belongs_to :pipeline_run
+
+  validates :gene, presence: true, if: :mass_validation_enabled?
+  validates :allele, presence: true, if: :mass_validation_enabled?
+  validates :drug_family, presence: true, if: :mass_validation_enabled?
+  # AmrCount is designed to handle partial data. See pipeline_run_spec.rb.
```

Comment:
@lvreynoso any elaboration welcome 🙂

Reply:
The code that loads amr count results for a pipeline run downloads the amr results file from s3 and then adds values for those keys (among others) only if they are present in the results table; they do not write a default value. It's possible that those attributes will not be present if something is wrong either with our data source (the ARGAnnot v2 fasta file) or the run_srst2 step in the pipeline. They shouldn't be null values when we try to save the AmrCount in rails (knowing what gene we detected is the whole point) but in theory it's possible.

I think it's good to validate the fields, as long as one model failing to save doesn't interfere with the rest, since the `AmrCounts` are saved in a batch. See https://github.com/chanzuckerberg/idseq-web/blob/8ae5dd19f3dee65da920ce6f9ef566d52c10b768/app/models/pipeline_run.rb#L643

Reply:
Thanks for the explanation. I adjusted to `allow_nil` for all columns in `amr_counts`.